### PR TITLE
Add guidance on how to deal with API breakages

### PR DESCRIPTION
**Pull request by philbull**
Fixes issue #308. Also tidies up the `CONTRIBUTE.md` file.

### CODE REVIEW

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -1,59 +1,72 @@
...
+    `check_ccl_SOURCES` in `Makefile.am`.
+ 5. Any other new files that should be included with the library (e.g. 
+    documentation files) should be listed under `EXTRA_DIST` in `Makefile.am`.
+ 6. The file `configure.ac` may have to be modified when adding new features 
+    (e.g. adding a new dependency).
+ 7. After making changes to any of these files, developers should run `autoreconf -i`.
+    and commit any modified files (some files will be automatically regenerated 
+    by this command).
+ 8. Make sure that you regenerate the SWIG Python wrapper (see below).
+ 9. Any updates to CCL should be documented in the CCL note that you can find
```
Comment (elisachisari) [inline CONTRIBUTE.md]: I think this should also be something to check for in the above list, when reviewing a PR.

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -1,59 +1,72 @@
...
+ 6. Look at the code (see "Files changed" on the top of the GitHub pull request 
+    page) and check that the changes make sense to you.
+ 7. If new science has been implemented, and if possible, try to compare the 
+    output of the code against your own predictions. Possibly, ask the developer 
+    to implement a unit test for it.
+ 8. Make sure that the libraries install successfully and the unit tests pass 
+    on Travis-CI.
+ 9. Make sure that the changes come with documentation, e.g. internally in the 
+    C code and through Python docstrings. Make sure that example code in the 
+    `examples/` directory has been updated appropriately.
```
Comment (elisachisari) [inline CONTRIBUTE.md]: And that the doxygen files have been replaced with the new ones?

**Comment by philbull** [comment]
Thanks @elisachisari, I've made these changes now.

**Comment by philbull** [comment]
I also renamed `CONTRIBUTE.md` to `CONTRIBUTING.md` so that GitHub recognises it (it expects to find a file with this name).

**Pull request merged**